src/spatialdata_plot/pl/render.py

```diff
@@ -456,8 +495,25 @@ def _render_points(
                 cmap=render_params.cmap_params.cmap,
             )
 
-        rbga_image = np.transpose(ds_result.to_numpy().base, (0, 1, 2))
-        cax = ax.imshow(rbga_image, zorder=render_params.zorder, alpha=render_params.alpha)
+        # create SpatialImage to get it back to original size
```

Comment:
Also this code is a duplicate, I'd consider it refactoring it into a private function.